XamarinCommunityToolkit/Effects/SafeAreaEffectRouter.ios.cs

```diff
@@ -14,34 +14,39 @@ public class SafeAreaEffectRouter : PlatformEffect
 	{
 		Thickness initialMargin;
 
+		new View Element
+			=> base.Element as View;
+
+		bool IsEligibleToConsumeEffect
+			=> Element != null
+				&& UIDevice.CurrentDevice.CheckSystemVersion(11, 0)
+				&& UIApplication.SharedApplication.Windows.Any();
+
 		protected override void OnAttached()
 		{
-			if (!(Element is View element))
-				return;
-
-			if (!UIDevice.CurrentDevice.CheckSystemVersion(11, 0) || !UIApplication.SharedApplication.Windows.Any())
+			if (!IsEligibleToConsumeEffect)
 				return;
 
 			var insets = UIApplication.SharedApplication.Windows[0].SafeAreaInsets;
-
-			initialMargin = element.Margin;
-
-			if (insets.Top <= 0)
```

Comment:
@sthewissen I am pretty sure this check is not needed (e.g. if some device has only left/right insets or only bottom etc.)